### PR TITLE
Added userIs() method to guards

### PR DESCRIPTION
**Pull request by stephanbouman**
More than once I compare the authenticated user against another user instance. It is code which repeats a lot. 

```php
if(auth()->user()->id === $post->user->id){
```
could be a lot easier to read:

```php
if(auth()->userIs($post->user)){
```
It makes the DX better with adding a bit syntactic sugar. We read actually what we are doing.

### CODE REVIEW

File: src/Illuminate/Contracts/Auth/Guard.php [hunk 1]
```diff
@@ -46,6 +48,13 @@ public function validate(array $credentials = []);
      * @return bool
      */
     public function hasUser();
+    
+    /**
+     * Compares the authenticated user with a given user.
+     *
+     * @return bool
+     */
+    public function userIs(AuthenticatableContract $user);
```
Comment (timacdonald) [inline src/Illuminate/Contracts/Auth/Guard.php:57]: The change to the contract makes this a breaking change if we are targeting 9.x

We could:

1. Target 10.x
2. Remove the contract change and just implement the method on the `GuardHelpers` trait to make it available on implementations.

Comment (BrandonSurowiec) [reply src/Illuminate/Contracts/Auth/Guard.php:57]: The Eloquent Model has an `is` method for this exact comparison:

````php
if (auth()->user()->is($post->user)) {
````

**Comment by BrandonSurowiec** [comment]
The Eloquent Model has an `is` method for this exact comparison:

````php
if (auth()->user()->is($post->user)) {
````

**Comment by stephanbouman** [comment]
Missed the is() method on the model. I was looking for it in the auth guard. Thanks!

**Pull request closed without merging**